portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsTextField.testcase

```diff
@@ -1270,6 +1270,113 @@ definition {
 			key_fieldReferenceValue = "Text1234");
 	}
 
+	@description = "LPS-129419 - Verify that the Required Field Error Message can be Edited"
+	@priority = "5"
+	test RequiredFieldErrorMessageCanBeEdited {
+		FormsAdmin.addForm();
+
+		Form.editName(formName = "Form 1");
+
+		Form.addTextField(fieldPositionNumber = "0");
+
+		FormFields.enableSwitch(fieldName = "Required");
+
+		FormFields.viewRequiredAsterisk();
+
+		AssertElementPresent(locator1 = "FormFields#SIDEBAR_BASIC_DEFAULT_ERROR_MESSAGE");
+
+		Uncheck.uncheckToggleSwitch(
+			key_toggleSwitchLabel = "Required",
+			locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
```

Comment:
@leticiachiu, there are some macros in `FormFields.macro` to perform the check and uncheck of the switches in Forms Sidebar, in light of this why did you prefer to use this external macro? Can you please clarify?

Reply:
I used this one because when I use `FormFields.disableSwitch` the test fails:
```
java.lang.Exception: Element is checked at "//input[@type='checkbox' and contains(@id,'Required') or contains (@name,'Required')] | //li[contains(.,'Required')]//input[@type='checkbox'] | //label[contains(@class,'toggle-switch')]//input[contains(@type,'checkbox')]/following-sibling::span[contains(@class,'toggle-switch-bar')]//span[contains(text(),'Required')]/..//preceding-sibling::input[contains(@type,'checkbox')] | //input[contains(@type,'checkbox')]//following-sibling::span[contains(.,'Required')]/..//preceding-sibling::input[contains(@type,'checkbox')]"
	at com.liferay.poshi.runner.selenium.BaseWebDriverImpl.assertNotChecked(BaseWebDriverImpl.java:528)
```

Reply:
@leticiachiu, apparently the problem is inside the macro `FormFields.viewCheckboxNotChecked(fieldName = "${fieldName}");` invoked by `FormFields.disableSwitch`. Is there any path associated to an unchecked switch which we can wait for before assert that the switch is unchecked? Can you please give it a try?

If you succeed on it then we won't need to use an external macro. :slightly_smiling_face: